data_management/rocrate.py

```diff
@@ -0,0 +1,661 @@
+
+    if len(user_authors) == 0:
+        agent_id = f"{registry_url}api/user/{code_run.updated_by.id}"
+        crate.add(
+            Person(
+                crate, agent_id, properties={"name": code_run.updated_by.full_name()}
+            )
+        )
+
+    else:
+        # we have an author linked to the user
+        agent_id = f"{registry_url}api/author/{user_authors[0].author.id}"
+        run_agent = crate.add(
+            Person(crate, agent_id, properties={"name": user_authors[0].author.name})
+        )
+        if user_authors[0].author.identifier is not None:
+            run_agent["identifier"] = user_authors[0].author.identifier
+
+    crate_code_run["agent"] = {"@id": agent_id}
+
+    # add the data product to the code run
+    crate_code_run["result"] = crate_data_product
+
+    return crate_code_run
+
+
+def _get_data_product(crate, data_product, registry_url, output):
+    """
+    Create an RO Crate file entity representing the data product.
+
+    @param crate_code_run: RO Crate entity representing the code run
+    @param data_product: a data_product from the DataProduct table
+    @param registry_url: a str containing the registry URL
+    @param output (bool): true if the data product is an output
+
+    @return an RO Crate file entity representing the data product
+
+    """
+    # Is it an external data product?
+    crate_data_product = _get_external_object(crate, data_product)
+
+    if crate_data_product is None:
+        crate_data_product = _get_local_data_product(
+            crate, data_product, registry_url, output
+        )
+
+    _add_licenses(crate, crate_data_product, data_product.object, registry_url)
+
+    _add_authors(
+        data_product.object.authors.all(),
+        crate,
+        crate_data_product,
+        registry_url,
+    )
+
+    return crate_data_product
+
+
+def _get_external_object(crate, data_product):
+    """
+    Create an RO Crate file entity for the given data product.
+
+    @param crate: the RO Crate object
+    @param data_product: a data_product from the DataProduct table
+
+    @return an RO Crate file entity representing the external object, may be None
+
+    """
+    # check for external object linked to the data product
+    try:
+        external_object = data_product.external_object
+    except (models.DataProduct.external_object.RelatedObjectDoesNotExist,):
+        return None
+
+    properties = {}
+    properties["name"] = external_object.title
+    properties["sdDatePublished"] = str(external_object.release_date)
+
+    if external_object.identifier:
+        source_loc = external_object.identifier
+    else:
+        source_loc = external_object.alternate_identifier
+
+    if external_object.description:
+        properties["description"] = external_object.description
+
+    if external_object.original_store:
+        properties["original_store"] = str(external_object.original_store)
```

Comment:
"original_store" is not a property in the RO-Crate context. I'm not sure what this represents. Maybe use [url](https://schema.org/url) or [contentUrl](https://schema.org/contentUrl)?

Reply:
`original_store` removed in https://github.com/FAIRDataPipeline/data-registry/pull/189/commits/a3ecb9741f8e9649a083318f4dcc30e36615c1f6